lnbits/wallets/zbd.py

```diff
@@ -0,0 +1,171 @@
+
+        self.endpoint = self.normalize_endpoint(settings.zbd_api_endpoint)
+        self.auth = {
+            "apikey": settings.zbd_api_key,
+            "User-Agent": settings.user_agent,
+        }
+        self.client = httpx.AsyncClient(base_url=self.endpoint, headers=self.auth)
+
+    async def cleanup(self):
+        try:
+            await self.client.aclose()
+        except RuntimeError as e:
+            logger.warning(f"Error closing wallet connection: {e}")
+
+    async def status(self) -> StatusResponse:
+        try:
+            r = await self.client.get("wallet", timeout=10)
+        except (httpx.ConnectError, httpx.RequestError):
+            return StatusResponse(f"Unable to connect to '{self.endpoint}'", 0)
+
+        if r.is_error:
+            error_message = r.json()["message"]
+            return StatusResponse(error_message, 0)
+
+        data = int(r.json()["data"]["balance"])
+        # ZBD returns everything as a str not int
+        # balance is returned in msats already in ZBD
+        return StatusResponse(None, data)
+
+    async def create_invoice(
+        self,
+        amount: int,
+        memo: Optional[str] = None,
+        description_hash: Optional[bytes] = None,
+        unhashed_description: Optional[bytes] = None,
+        **kwargs,
+    ) -> InvoiceResponse:
+        # https://api.zebedee.io/v0/charges
+        if description_hash or unhashed_description:
+            raise Unsupported("description_hash")
+
+        msats_amount = amount * 1000
+        data: Dict = {
+            "amount": f"{msats_amount}",
+            "description": memo,
+            "expiresIn": 3600,
+            "callbackUrl": "",
+            "internalId": "",
+        }
+
+        r = await self.client.post(
+            "charges",
+            json=data,
+            timeout=40,
+        )
+
+        if r.is_error:
+            error_message = r.json()["message"]
+            return InvoiceResponse(False, None, None, error_message)
+
+        data = r.json()["data"]
+        checking_id = data["id"]  # this is a zbd id
+        payment_request = data["invoice"]["request"]
+        return InvoiceResponse(True, checking_id, payment_request, None)
+
+    async def pay_invoice(self, bolt11: str, fee_limit_msat: int) -> PaymentResponse:
+        # https://api.zebedee.io/v0/payments
+        r = await self.client.post(
+            "payments",
+            json={
+                "invoice": bolt11,
+                "description": "",
+                "amount": "",
+                "internalId": "",
+                "callbackUrl": "",
+            },
+            timeout=None,
+        )
+
+        if r.is_error:
+            error_message = r.json()["message"]
+            return PaymentResponse(False, None, None, None, error_message)
+
+        data = r.json()
+
+        # get the payment hash from the zbd api
+        decoded_request = await self.client.post(
```

Comment:
Can we avoid this second HTTP call?
The previous HTTP call (https://zbd.dev/api-reference/payments/send) provides us the `preimage` and the `fee`.  The payment hash can be extracted using:

```py
from lnbits import bolt11
invoice = bolt11.decode(bolt11)
# invoice.payment_hash
```

Reply:
OK.